kornia/contrib/models/rt_detr/model.py

```diff
@@ -0,0 +1,186 @@
+
+        elif model_type == RTDETRModelType.resnet101:
+            backbone = ResNetD.from_config(101)
+            neck_hidden_dim = config.neck_hidden_dim or 384
+            neck_dim_feedforward = config.neck_dim_feedforward or 2048
+            head_num_decoder_layers = config.head_num_decoder_layers or 6
+            neck_expansion = config.neck_expansion or 1.0
+
+        elif model_type == RTDETRModelType.hgnetv2_l:
+            backbone = PPHGNetV2.from_config("L")
+            neck_hidden_dim = config.neck_hidden_dim or 256
+            neck_dim_feedforward = config.neck_dim_feedforward or 1024
+            head_num_decoder_layers = config.head_num_decoder_layers or 6
+            neck_expansion = config.neck_expansion or 1.0
+
+        elif model_type == RTDETRModelType.hgnetv2_x:
+            backbone = PPHGNetV2.from_config("X")
+            neck_hidden_dim = config.neck_hidden_dim or 384
+            neck_dim_feedforward = config.neck_dim_feedforward or 2048
+            head_num_decoder_layers = config.head_num_decoder_layers or 6
+            neck_expansion = config.neck_expansion or 1.0
+
+        model = RTDETR(
+            backbone,
+            HybridEncoder(backbone.out_channels, neck_hidden_dim, neck_dim_feedforward, neck_expansion),
+            RTDETRHead(
+                config.num_classes,
+                config.head_hidden_dim,
+                config.head_num_queries,
+                [neck_hidden_dim] * 3,
+                head_num_decoder_layers,
+            ),
+        )
+
+        if config.checkpoint:
+            model.load_checkpoint(config.checkpoint)
+        return model
+
+    def forward(self, images: Tensor) -> Tensor:
+        """Detect objects in an image.
+
+        Args:
+            images: images to be detected. Shape :math:`(N, C, H, W)`.
+
+        Returns:
+            Object detection results, in format class id, score, bounding box xywh. Shape :math:`(N, D, 6)`,
+            where :math:`D` is the number of detections.
+        """
+        H, W = images.shape[2:]
+        fmaps = self.backbone(images)
+        fmaps = self.neck(fmaps)
+        bboxes, logits = self.head(fmaps)
+
+        # https://github.com/PaddlePaddle/PaddleDetection/blob/5d1f888362241790000950e2b63115dc8d1c6019/ppdet/modeling/post_process.py#L446
+        # box format is cxcywh
+        # convert to xywh
+        # bboxes[..., :2] -= bboxes[..., 2:] * 0.5  # in-place operation is not torch.compile()-friendly
+        cxcy = bboxes[..., :2]
+        wh = bboxes[..., 2:]
+        bboxes = concatenate([cxcy - wh * 0.5, wh], -1)
+
+        bboxes = bboxes * tensor([W, H, W, H], device=bboxes.device, dtype=bboxes.dtype).view(1, 1, 4)
+        scores = logits.sigmoid()  # RT-DETR was trained with focal loss. thus sigmoid is used instead of softmax
+
+        # the original code is slightly different
+        # it allows 1 bounding box to have multiple classes (multi-label)
+        scores, labels = scores.max(-1)
```

Comment:
one last thought -- would that work when we want to fine tune ? the max could be part of the process of ObjectDetectionResults ? @johnnv1

Reply:
i think so, all the preprocessing maybe should go there or maybe in the high-level API